vochain/scrutinizer/db.go

```diff
@@ -83,6 +82,33 @@ func (r *Results) String() string {
 	}
 	return results.String()
 }
+func (r *Results) Add(new *Results) error {
+	r.Weight.Add(r.Weight, new.Weight)
```

Comment:
This line adds the old results weight to the new partial results weight, whereas the old corresponding line simply sets the new weight. 
```
update.Weight.Set(results.Weight)
```

I remember we had a bug with this in the past that caused the weight to increase exponentially. Is this change (along with the renaming of `results` to `partialResults` clarifying that commitVotes() should use a partial results which will be added to, rather than replace, the current results?

Reply:
AAAhhh, now I understand! I was crazy trying to understand why commitVotes was doing Set instead of Add :)

The current description of commitVotes is as follows (it was actually like this, I did not modify it). Would you make it more clear? Can you propose an alternative? Thanks.
```
// commitVotes adds the votes and weight from results to the local database.
// It does not overwrite the stored results but update them by adding the new content.
func (s *Scrutinizer) commitVotes(pid []byte, partialResults *Results, height uint32) error 
```

Reply:
Btw @NateWilliams2 I think the actual bug was not in the commitVotes() method but in the scrutinizer_test, which I fixed on this PR.

Reply:
I see! No, this looks clear to me. I "fixed" the bug according to the tests, rather than the specification for the method, assuming that the test was correct. Sounds like this is cleared up now :)